diff -Naurp kodi/xbmc/cores/VideoPlayer/DVDMessageQueue.cpp kodi-patch/xbmc/cores/VideoPlayer/DVDMessageQueue.cpp
--- kodi/xbmc/cores/VideoPlayer/DVDMessageQueue.cpp	2024-05-12 07:45:35.665356582 +0000
+++ kodi-patch/xbmc/cores/VideoPlayer/DVDMessageQueue.cpp	2024-05-12 07:47:03.057082436 +0000
@@ -307,7 +307,7 @@ void CDVDMessageQueue::WaitUntilEmpty()
   }
 }
 
-int CDVDMessageQueue::GetLevel(bool data_level) const
+int CDVDMessageQueue::GetLevel() const
 {
   std::unique_lock<CCriticalSection> lock(m_section);
 
@@ -316,9 +316,9 @@ int CDVDMessageQueue::GetLevel(bool data
   if (m_iDataSize == 0)
     return 0;
 
-  if (IsDataBased() || data_level)
+  if (IsDataBased())
   {
-    return std::min((uint64_t)100, 100 * m_iDataSize / m_iMaxDataSize);
+    return std::min(100, 100 * m_iDataSize / m_iMaxDataSize);
   }
 
   int level = std::min(100.0, ceil(100.0 * m_TimeSize * (m_TimeFront - m_TimeBack) / DVD_TIME_BASE ));
diff -Naurp kodi/xbmc/cores/VideoPlayer/DVDMessageQueue.h kodi-patch/xbmc/cores/VideoPlayer/DVDMessageQueue.h
--- kodi/xbmc/cores/VideoPlayer/DVDMessageQueue.h	2024-05-12 07:45:35.665356582 +0000
+++ kodi-patch/xbmc/cores/VideoPlayer/DVDMessageQueue.h	2024-05-12 07:47:03.057082436 +0000
@@ -80,8 +80,8 @@ public:
   void WaitUntilEmpty();
 
   // non messagequeue related functions
-  bool IsFull() const { return GetLevel(true) == 100; }
-  int GetLevel(bool data_level = false) const;
+  bool IsFull() const { return GetLevel() == 100; }
+  int GetLevel() const;
 
   void SetMaxDataSize(int iMaxDataSize) { m_iMaxDataSize = iMaxDataSize; }
   void SetMaxTimeSize(double sec) { m_TimeSize  = 1.0 / std::max(1.0, sec); }
@@ -102,14 +102,15 @@ private:
   bool m_bInitialized;
   bool m_drain = false;
 
-  uint64_t m_iDataSize;
+  int m_iDataSize;
   double m_TimeFront;
   double m_TimeBack;
   double m_TimeSize;
 
-  uint64_t m_iMaxDataSize;
+  int m_iMaxDataSize;
   std::string m_owner;
 
   std::list<DVDMessageListItem> m_messages;
   std::list<DVDMessageListItem> m_prioMessages;
 };
+
